inverse_covariance/two_way_standard_scaler.py

```diff
@@ -0,0 +1,329 @@
+                delattr(self, attr)
+
+    def _initial_state(self):
+        return not hasattr(self, "n_rows_seen_")
+
+    def fit(self, X, y=None):
+        """Compute the mean and std for both row and column dimensions.
+
+        Parameters
+        ----------
+        X : {array-like}, shape [n_rows, n_cols]
+            The data used to compute the mean and standard deviation
+            along both row and column axes
+
+        y : Passthrough for ``Pipeline`` compatibility. Input is ignored.
+        """
+        # Reset internal state before fitting
+        self._reset()
+        return self.partial_fit(X, y)
+
+    def partial_fit(self, X, y=None):
+        """Online computation of mean and std on X for later scaling."""
+        X = check_array(
+            X,
+            accept_sparse=False,
+            copy=self.copy,
+            warn_on_dtype=True,
+            dtype=FLOAT_DTYPES,
+            estimator=self,
+            ensure_min_features=2,
+        )
+        if sparse.issparse(X):
+            raise NotImplemented(
+                "Input is sparse: Algorithm for sparse matrices currently not supported."
+            )
+
+        self.n_rows_, self.n_cols_ = np.shape(X)
+
+        # Q: This doesnt seem to actually get used in the transform, only in
+        #    the inverse transform which it sounds like we should not support.
+
+        # initialize variables on first pass
+        if self._initial_state():
+            self.col_mean_ = 0.
+            self.col_var_ = 0. if self.with_std else None
+            self.n_cols_seen_ = 0
+            self.row_mean_ = 0.
+            self.row_var_ = 0. if self.with_std else None
+            self.n_rows_seen_ = 0
+
+        self.col_mean_, self.col_var_, self.n_rows_seen_ = _incremental_mean_and_var(
+            X, self.col_mean_, self.col_var_, self.n_rows_seen_
+        )
+        self.row_mean_, self.row_var_, self.n_cols_seen_ = _incremental_mean_and_var(
+            X.T, self.row_mean_, self.row_var_, self.n_cols_seen_
+        )
+
+        self.row_scale_ = None
+        self.col_scale_ = None
+        if self.with_std:
+            self.row_scale_ = _handle_zeros_in_scale(np.sqrt(self.row_var_))
+            self.col_scale_ = _handle_zeros_in_scale(np.sqrt(self.col_var_))
+
+        return self
+
+    def transform(self, X, copy=False):
+        """Perform standardization by centering and scaling
+        Parameters
+        ----------
+        X : array-like, shape [n_rows, n_cols]
+            The data used to scale along the features axis.
+        copy : bool, optional (default: False)
+            Copy the input X or not.
+        """
+        check_is_fitted(self, "row_scale_")
+        X = check_array(
+            X,
+            accept_sparse=False,
+            copy=copy,
+            warn_on_dtype=True,
+            dtype=FLOAT_DTYPES,
+            estimator=self,
+        )
+        n_rows, n_cols = np.shape(X)
+        if self.n_cols_ != n_cols:
```

Comment:
This check was required to pass the sklearn `check_estimator` suite.  The main idea is that we throw a value error if the dimention of the transform data is not the same as that of the fit method.  Does this seem right?

Reply:
Yeah this sounds good.